rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -261,7 +264,8 @@ rmw_destroy_client(rmw_node_t * node, rmw_client_t * client)
     }
     delete info;
   }
-  delete (client);
+  rmw_free(const_cast<char *>(client->service_name));
```

Comment:
should we check for if service_name is null before calling rmw_free on it ? (like its done above) Same below